### ISSUE TITLE
[Bug] Unable to connect to dev EC2 instance

### ISSUE THREAD

**Issue by tschaffter** [Sage-Bionetworks/sage-monorepo#2484]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What product(s) are you seeing the problem on?

Sage Monorepo

### Current behavior

I just updated VS Code:

```
Version: 1.86.0 (Universal)
Commit: 05047486b6df5eb8d44b2ecd70ea3bdf775fd937
```

I can no longer connect to the EC2 instance used for development:

```
The remote host may not meet VS Code Server's prerequisites for glibc and libstdc++
```

### Are you developing inside the dev container?

- [X] I am using the dev container

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tschaffter** [comment]
<img width="454" alt="image" src="https://github.com/Sage-Bionetworks/sage-monorepo/assets/3056480/01b83079-e37e-44e4-b193-8dbef1f74d5f">

**Comment by tschaffter** [comment]
1. Ssh to the instance with the terminal (VS Code won't connect)
2. Update the OS packages: `sudo yum update`

Restarting the EC2 and VS Code didn't help

**Comment by tschaffter** [comment]
- Here are the [requirements](https://code.visualstudio.com/docs/remote/linux#_remote-host-container-wsl-linux-prerequisites) for VS Code.

My kernel version is below the min version required by VS Code.

```
$ uname -r 
4.14.336-253.554.amzn2.x86_64
```

Glibc version:

```
$ ldd --version
ldd (GNU libc) 2.26
```

The kernel and Glibc versions are below the required versions.

**Comment by tschaffter** [comment]
Creating a new instance:

- Our docs says `t3.2xlarge` which is no longer available in AWS Console

**Comment by tschaffter** [comment]
Requested the type `t3.2xlarge` to be added back to the SC in this [Jira](https://sagebionetworks.jira.com/browse/IT-3222) ticket.

**Comment by tschaffter** [comment]
## Workaround

Downgrading VS Code to the previous version (1.85.2) solves the issue.

**Comment by tschaffter** [comment]
## Update

- The issue affect Mac and Windows
- Creating a new EC2 instance with the SC does not solve the issue
  - Instance name: tschaffter-devcontainers-2024Q1
  - SC instance type: EC2: Linux Docker
  - SC instance version: v1.2.12
  - EC2 instance type: t3a.2xlarge
  - Other info
      ```
      $ uname -r
      4.14.320-242.534.amzn2.x86_64
      
      $ldd --version
      ldd (GNU libc) 2.26
      Copyright (C) 2017 Free Software Foundation, Inc.
      ```

**Comment by tschaffter** [comment]
## Benchmarking

### t3a.2xlarge

```
$ openchallanges-build-images

real    11m16.691s
user    9m19.836s
sys     0m49.420s
```

**Comment by tschaffter** [comment]
## Migrating to `t3a.2xlarge`

I was able to deploy the OC stack there and access the web app. We can start using this instance instead of `t3.2xlarge`.

**Comment by tschaffter** [comment]
## Upgrade Amazon Linux 2 kernel version to v5.15

```
$ sudo amazon-linux-extras |grep kernel
 49  kernel-5.4               available    [ =stable ]
 55  kernel-5.10              available    [ =stable ]
 62  kernel-5.15              available    [ =stable ]
```

**Comment by tschaffter** [comment]
## Update

This ticket is blocked until an EC2 instance with AL2023 can be created with the Service Catalog.

**Comment by vpchung** [comment]
^ I additionally needed to downgrade my version of Remote-SSH, as the latest version is only compatible with VSCode 1.86.0

**Comment by tschaffter** [comment]
[Update] To all VS Code users who use Remote Development (SSH, WSL, Dev Container):
VS Code released a new version (1.86.1) that addresses the [issue](https://sagebionetworks.slack.com/archives/CMSGC6ATD/p1707333269504689) previously reported in this channel :point_up::skin-tone-2:. With this update, you will be able to develop remotely again with SSH, WSL and/or Dev Container.

**Comment by tschaffter** [comment]
I will still be tracking the Jira ticket I opened to request an AL2023 image that will meet the requirement of VS Code. For now we will use what VS Code calls their "legacy server" on OS that don't meet the new requirements.
